### DISCUSSION TITLE
Fastdds DiscoveryServer Mode  Too many open files

### DISCUSSION THREAD

**Discussion by kaola-zero** [eProsima/Fast-DDS#5456] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I encountered three problems：
First:
I use FastDDS-2.14.0 version,I used the [Simple Discovery]method and found that establishing 100 publish and subscribe connections takes about 35 seconds, which is a bit long. Is there any way to reduce the time? 5 seconds is ideal, and 10 seconds is acceptable
second:
When I use the discovery server mode in the routine with 100 subscriptions and 100 publishers, the following error occurs: “terminate called after throwing an instance of 'td:: system_ error'
what():   eventfd_select_interrupter: Too many open files
Abandoned (core has been dumped)”, will files always be opened when creating client publish and subscribe? I tried it and found that creating 50 subscriptions requires opening 872 files. Is there a way not to create files？
third:
I increased the number of files that can be opened by changing the system configuration, using the same discovery server mode. However, I found that when the number of published and subscribed files exceeds 50, the connection cannot be established properly.Is there any way to increase the number of connections？
                                                                                                                                                    Thank you for any answers！

### Current behavior

First:
I use FastDDS-2.14.0 version,I used the [Simple Discovery]method and found that establishing 100 publish and subscribe connections takes about 35 seconds, which is a bit long. Is there any way to reduce the time? 5 seconds is ideal, and 10 seconds is acceptable
second:
When I use the discovery server mode in the routine with 100 subscriptions and 100 publishers, the following error occurs: “terminate called after throwing an instance of 'td:: system_ error'
what():   eventfd_select_interrupter: Too many open files
Abandoned (core has been dumped)”, will files always be opened when creating client publish and subscribe? I tried it and found that creating 50 subscriptions requires opening 872 files. Is there a way not to create files？
third:
I increased the number of files that can be opened by changing the system configuration, using the same discovery server mode. However, I found that when the number of published and subscribed files exceeds 50, the connection cannot be established properly.Is there any way to increase the number of connections？
                                                                                                                                                    Thank you for any answers！

### Steps to reproduce

Use DiscoveryServerExample build 100 sub or pub

### Fast DDS version/commit

Fast-DDS-2.14.0

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

UDPv4

**Comment by cferreiragonz** [comment]
Hi @kaola-zero! Thanks for using Fast DDS.

It would be very helpful if you could provide more details about how you are configuring your tests. Do you have a `.patch` file with the modifications made to the Discovery Server example? This would allow us to try reproducing the issue on our side.

Please note that the number of files created per process is not directly related to the number of subscribers or publishers in a participant. Therefore, increasing the number of subscribers or publishers should not result in an increase in the number of files used by the process. Additionally, there is currently no limitation on the number of connections supported by the Discovery Server.

Lastly, I am moving this issue to the corresponding discussion forum, as Ubuntu 20 is no longer supported according to [PLATFORMS_SUPPORT.md](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md#platforms).
